### PR TITLE
Feature/issue 19 maven artifact as root module

### PR DESCRIPTION
**Pull request by Clayton7510**

### CODE REVIEW

File: tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformGetMavenRootArtifact.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    this.tfRootDir = tfRootDir;
+  }
+
+  @Override
+  public String execute(Properties properties) throws TerraformException {
+    getArtifactFromMavenRepo(new DefaultInvoker(), new DefaultInvocationRequest());
+    Path expandedArtifactPath = expandMavenArtifacts();
+    try {
+      String userDir = System.getProperty("user.dir");
+      synchronized (this.getClass()) { //
```
Comment (Clayton7510) [inline tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformGetMavenRootArtifact.java]: why is this synchronized?

Comment (Clayton7510) [inline tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformGetMavenRootArtifact.java]: why is this synchronized?

File: tf-build-tools/tf-maven-plugin/pom.xml [hunk 2]
```diff
@@ -15,6 +15,7 @@
     <maven.install.skip>false</maven.install.skip>
     <gpg.skip>false</gpg.skip>
     <skipNexusStagingDeployMojo>false</skipNexusStagingDeployMojo>
+    <excludedGroups>com.deliveredtechnologies.test.categories.Integration</excludedGroups>
```
Comment (Clayton7510) [inline tf-build-tools/tf-maven-plugin/pom.xml]: This excludes Integration test group(s) by default.
The Integration test group includes tests that require external bootstrapping.

File: tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformGet.java [hunk 3]
```diff
@@ -40,12 +42,12 @@
    * @param tfModules the common modules directory; if null, it's defaulted to src/main/.tfmodules
    */
   public TerraformGet(Log log, String tfModules) throws IOException {
-    this(log, TerraformUtils.getDefaultTfModulesDir());
+    this(log, StringUtils.isEmpty(tfModules) ? null : Paths.get(tfModules));
```
Comment (Clayton7510) [inline tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformGet.java]: This is a bug fix. Previously, when tfModules directory was supplied, it was ignored.

File: tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformGet.java [hunk 4]
```diff
@@ -40,12 +42,12 @@
...
   }
 
   protected TerraformGet(Log log, Path tfModules) throws IOException {
     this.log = log;
-    this.tfModules = tfModules;
+    this.tfModules = tfModules == null ? TerraformUtils.getDefaultTfModulesDir() : tfModules;
```
Comment (Clayton7510) [inline tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformGet.java]: A continuation of the fix from above.

File: tf-build-tools/tf-maven-plugin/src/test/java/com/deliveredtechnologies/maven/terraform/TerraformGetMavenRootArtifactIntTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    String artifact = "com.deliveredtechnologies.example.maven.tf:tf-s3-consumer:1.0";
+    Path tfWorkingPath = Paths.get(this.workingDir, ".tfproject");
+    Path tfModulesPath = Paths.get(this.workingDir, ".tfproject", "src", "main", ".tfmodules");
+
+    try {
+      TerraformGetMavenRootArtifact terraformGetMavenRootArtifact = new TerraformGetMavenRootArtifact(artifact, new Slf4jMavenAdapter(LoggerFactory.getLogger(this.getClass())));
+      terraformGetMavenRootArtifact.execute(new Properties());
+      Assert.assertTrue(tfWorkingPath.resolve("src").resolve("main").resolve("tf").resolve("s3-consumer").resolve("main.tf").toFile().exists());
+      Assert.assertTrue(tfModulesPath.resolve("s3").resolve("s3_replicated_src").resolve("main.tf").toFile().exists());
+      Assert.assertTrue(tfModulesPath.resolve("s3").resolve("s3").resolve("main.tf").toFile().exists());
```
Comment (s1234a6i) [inline tf-build-tools/tf-maven-plugin/src/test/java/com/deliveredtechnologies/maven/terraform/TerraformGetMavenRootArtifactIntTest.java]: check for pom file

Comment (s1234a6i) [inline tf-build-tools/tf-maven-plugin/src/test/java/com/deliveredtechnologies/maven/terraform/TerraformGetMavenRootArtifactIntTest.java]: check for pom file

**Pull request merged**